File: Data Structures & Algorithms/binary-search/submission-1.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size() - 1;
        int m = nums.size() / 2;

        while (l <= r) {
            int item = nums[m];

            if (target == item) {
                return m;
            }
            if (target < item) {
                r = m - 1;
            }
            if (target > item) {
                l = m + 1;
            }
            m = (r + l) / 2;
        }
        return -1;
    }
};

File: Data Structures & Algorithms/jump-game/submission-1.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        if len(nums) == 1:
            return True
        target = len(nums) - 1
        canMake = False
        for i in range(len(nums) - 2, -1, -1):
            val = nums[i]
            if val < target - i:
                canMake = False
            else:
                canMake = True
                target = i
        return canMake